src/main/kotlin/io/arrowkt/tagless/io/app/app.kt

```diff
@@ -0,0 +1,132 @@
+package io.arrowkt.tagless.io.app
+
+import arrow.core.none
+import arrow.core.some
+import arrow.data.ReaderT
+import arrow.data.extensions.kleisli.monad.monad
+import arrow.data.fix
+import arrow.effects.ForIO
+import arrow.effects.IO
+import arrow.effects.extensions.io.monadError.monadError
+import arrow.effects.fix
+import io.arrowkt.tagless.repository.AccountRepository
+import io.arrowkt.tagless.repository.interpreter.AccountRepositoryInMemory
+import io.arrowkt.tagless.service.AccountType
+import io.arrowkt.tagless.service.interpreter.AccountServiceInterpreter
+import io.arrowkt.tagless.service.interpreter.InterestPostingServiceInterpreter
+import io.arrowkt.tagless.service.interpreter.ReportingServiceInterpreter
+
+val accountServiceIO = AccountServiceInterpreter(IO.monadError())
+val interestPostingServiceIO = InterestPostingServiceInterpreter(IO.monadError())
+val reportingServiceIO = ReportingServiceInterpreter(IO.monadError())
+val monad = ReaderT.monad<ForIO, AccountRepository<ForIO>>(IO.monadError())
+
+fun main() {
```

Comment:
I wouldn't subscribe them separately. I'd make each function return an IO, merge them here, and use a single exit point for all.

In the future it'll be `suspend fun main()` so you'd just make sure they're started independently, i.e. Fiber.

Reply:
Just following the code from the book :smile:

Reply:
Hm.. but then it's different than the IO sample... do I also change that?

Reply:
I decided not to do this.

I know that is wrong and ideally you have no more that 1 `unsafeRunXXX` per app. However I want to be as close as possible as the original sample, where we have exactly [this](https://github.com/debasishg/frdomain-extras/blob/cd6793701761bf3b3144883a1b6a623654292563/tagless/src/main/scala/frdomain/ch6/domain/io/app/app.scala).

There is even duplication in subscribing to each use case:
```
    y.single.subscribe(
        { it.forEach(::println) },
        { it.printStackTrace() }
    )
```
this code is repeated for each use case.

And the even bigger problem:
```
override fun store(a: Account): Kind<F, Account> = catch {
        repo[a.no] = a
        a
    }
```
this code is using `MonadError` instead of `MonadDefer`.

But all this is consistent with the original code. People will be confused by the difference in syntax, by having to pass around typeclass instances, let's not confuse them even further by additionally altering the sample.